src/libmongoc/tests/test-mongoc-secure-channel.c

```diff
@@ -0,0 +1,263 @@
+#include <mongoc/mongoc.h>
+
+#include "TestSuite.h"
+#include "test-libmongoc.h"
+
+#if defined(MONGOC_ENABLE_SSL_SECURE_CHANNEL)
+#include <mongoc/mongoc-stream-tls-secure-channel-private.h>
+#include <mongoc/mongoc-host-list-private.h> // _mongoc_host_list_from_string_with_err
+#include <mongoc/mongoc-client-private.h>    // mongoc_client_connect_tcp
+#include <mongoc/mongoc-log-private.h>       // _mongoc_log_get_handler
+#include <test-conveniences.h>
+#include <mlib/test.h>
+
+static bool
+connect_with_secure_channel_cred (mongoc_ssl_opt_t *ssl_opt, mongoc_shared_ptr cred_ptr, bson_error_t *error)
```

Comment:
All calls to `connect_with_secure_channel_cred` are followed by an assertion that the result is true. Consider changing return type from `bool` to `void` and move assertions into this function.